lib/send-json.js

```diff
@@ -5,8 +5,14 @@
 
 'use strict';
 
+var escapeHtml = require('escape-html');
+
 module.exports = function sendJson(res, data) {
-  var content = JSON.stringify({error: data});
+  var content = JSON.stringify({error: data}, function(key, value) {
+    return typeof value  === 'string' && key !== 'stack' ?
+      escapeHtml(value) :
```

Comment:
I am afraid this does not make any sense to me. The JSON is a transport format that can be used in different ways - store it in SQL (think of a table storing all failed responses for future inspection), render it as HTML from a single-page app, etc. It's the responsibility of the code consuming this data to apply any escaping needed, depending on how it works with that data.

Please revert this change.

Reply:
@bajtos This may not be the right fix, but it is the type of fix that we actually wanted make because it is a low level security issue by pushing out potentially hazardous text to the frontend (I just did the HTML to be nice). I agree it is the developer's responsibility to ensure proper HTML escaping, but shouldn't this library (or loopback.io) be cautious of what it returns?

Reply:
In the world of API Economy, there are more and more API consumers that are not HTML websites. Consider for example a conversational bot invoking LoopBack APIs in background. Would you like to see HTML-escaped error messages in your Skype chat session because LoopBack is assuming all JSON consumers are web apps? If this example looks too silly, then consider a microservice invoking a LoopBack backend and reporting errors returned by LoopBack to a text log file. I would not want to see HTML-escaped text there, because it's unnecessarily difficult to read.

Don't get me wrong, I understand that in your particular case it may be helpful (or at least tempting) to escape string values as early as possible. But I still think escaping JSON values is not really solving the security problem. If you don't trust your HTML front-end is correctly escaping string values, then you should be HTML-escaping strings in all JSON responses, including those fetching data from your database. And if your database is SQL-based and we apply the same logic, then you need to SQL-encode all input values coming from your HTML front-ends too, to make sure they don't accidentally trigger SQL-injection attack in your backend.

Anyways, I don't mind supporting such setup by allowing strong-error-handler users to provide a custom function for producing JSON response bode, there is already a work in progress in #67.

However, the default behavior of strong-error-handler should stay as it is now, i.e. no HTML-escaping in JSON responses.